mlos_bench/mlos_bench/environments/script_env.py

```diff
@@ -92,3 +99,22 @@ def _get_env_params(self) -> Dict[str, str]:
             rename.update(self._shell_env_params_rename)
 
         return {key_sub: str(self._params[key]) for (key_sub, key) in rename.items()}
+
+    def _extract_stdout_results(self, stdout: str) -> Dict[str, float]:
+        """
+        Extract the results from the stdout of the script.
+
+        Parameters
+        ----------
+        stdout : str
+            The stdout of the script.
+
+        Returns
+        -------
+        results : Dict[str, float]
+            A dictionary of results extracted from the stdout.
+        """
+        if not self._parse_results_stdout:
+            return {}
+        _LOG.debug("Extract regex: '%s' from: '%s'", self._parse_results_stdout, stdout)
+        return {key: float(val) for (key, val) in self._parse_results_stdout.findall(stdout)}
```

Comment:
A thought: this will only support full word matches, correct?

Are groupings or replacements possible?

For instance (in Perl psuedocode) 

```python
   pat = re.sub(r'(foo)?bar(baz)?', r'\1\2')
   # returns one of foobaz, foo, baz
   # whereas findall() would return foobar, barbaz, foobarbaz
```

I don't know what type of output you have in mind for this, but I can imagine cases where there are extra characters in the output that we want to stip and reformat.

Could even be an array of those search/replace patterns actually.

Reply:
yes, we can easily strip the extra characters. In fact, this is the regex that I tested on our actual output of SysBench+MySQL:
```python
In [5]: re.findall(r"results-cumulative,[^,]*,[^,]*,(\w+),([0-9.]+)", stdout)
Out[5]:
[('time', '600.095825'),
 ('threads', '0'),
 ('events', '2074.051638'),
 ('total', '2074.051638'),
 # . . . .
]
```
where `stdout` looks like
```
Enable succeeded:
[stdout]
results-intermediate,2023-08-30 00:13:51,560.00,latency_pct,66.838153
results-intermediate,2023-08-30 00:13:51,560.00,errors,0.000000
. . . . . . .
results-cumulative,2023-08-30 00:14:31,600.10,time,600.095825
results-cumulative,2023-08-30 00:14:31,600.10,threads,0
results-cumulative,2023-08-30 00:14:31,600.10,events,2074.051638
results-cumulative,2023-08-30 00:14:31,600.10,total,2074.051638
. . . . . . .
```
that is, `re.findall()` extracts all tuples of regex groups from the text. The only thing we _cannot_ do is reordering of the groups, but I would not bother about it now. After all, this is intended to be a quick solution that a) fixes a bug that @eujing had discovered, and b) gives an ability to quickly extract the data from stdout without using additional scripts or an extra LocalEnv. We can always add more functionality later if we really need to

Reply:
oh, sorry I have not immediately realized what you meant. Yes, technically the user can write a regex that would have multiple overlapping matches. I don't really know how to make it foolproof - I would rather assume that the user really knows what they are doing :)

Reply:
Would be good to file that bug to give more context.